test/integration/infrastructure/infrastructure_test.go

```diff
@@ -193,24 +193,39 @@ var _ = Describe("Infrastructure tests", func() {
 
 			networkName := namespace
 			cloudRouterName := networkName + "-cloud-router"
-
-			err = prepareNewNetwork(ctx, log, project, computeService, networkName, cloudRouterName)
-			Expect(err).NotTo(HaveOccurred())
+			ipAddressNames := []string{networkName + "-manual-nat1", networkName + "-manual-nat2"}
 
 			var cleanupHandle framework.CleanupActionHandle
 			cleanupHandle = framework.AddCleanupAction(func() {
 				err := teardownNetwork(ctx, log, project, computeService, networkName, cloudRouterName)
 				Expect(err).NotTo(HaveOccurred())
+				err = teardownIPAddresses(ctx, log, project, computeService, ipAddressNames)
+				Expect(err).NotTo(HaveOccurred())
 
 				framework.RemoveCleanupAction(cleanupHandle)
 			})
 
-			providerConfig := newProviderConfig(&gcpv1alpha1.VPC{
+			err = prepareNewNetwork(ctx, log, project, computeService, networkName, cloudRouterName)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = prepareNewIPAddresses(ctx, log, project, computeService, ipAddressNames)
+			Expect(err).NotTo(HaveOccurred())
+
+			vpc := &gcpv1alpha1.VPC{
 				Name: networkName,
 				CloudRouter: &gcpv1alpha1.CloudRouter{
 					Name: cloudRouterName,
 				},
-			})
+			}
+			var natIPNames []gcpv1alpha1.NatIPName
+			for _, ipAddressName := range ipAddressNames {
+				natIPNames = append(natIPNames, gcpv1alpha1.NatIPName{Name: ipAddressName})
+			}
+			cloudNAT := &gcpv1alpha1.CloudNAT{
+				MinPortsPerVM: pointer.Int32(2048),
```

Comment:
Do we have to set it as part of https://github.com/gardener/gardener-extension-provider-gcp/issues/317?
You can either remove this or add assertions that verify later that this field is reflected in the created infrastructure on the cloud provider side.